cmd/manager/main.go

```diff
@@ -18,128 +18,45 @@ package main
 
 import (
 	"flag"
-	"fmt"
-	"net"
 	"os"
-	"os/signal"
-	"time"
 
 	"github.com/go-logr/logr"
-	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client/config"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 
-	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/controller"
+	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/manager"
 	poolmanager "github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/pool-manager"
-	"github.com/K8sNetworkPlumbingWG/kubemacpool/pkg/webhook"
 )
 
-var log logr.Logger
-var metricsAddr string
-var restartChannel = make(chan struct{})
-
-func loadMacAddressFromEnvVar(envName string) net.HardwareAddr {
-	if value, ok := os.LookupEnv(envName); ok {
-		poolRange, err := net.ParseMAC(value)
-		if err != nil {
-			log.Error(err, "Failed to parse mac address", "mac address", value)
-			os.Exit(1)
-		}
-
-		return poolRange
-	}
-
-	log.Error(fmt.Errorf("Environment variable %s don't exist", envName), "")
-	os.Exit(1)
-
-	// with out the the compilation failed
-	return nil
-}
-
-func checkForKubevirt(kubeClient *kubernetes.Clientset) bool {
-	result := kubeClient.ExtensionsV1beta1().RESTClient().Get().RequestURI("/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/virtualmachines.kubevirt.io").Do()
-	if result.Error() == nil {
-		return true
-	}
-
-	return false
-}
-
-// Check for Kubevirt CRD to be available
-func waitForKubevirt(kubeClient *kubernetes.Clientset) {
-	for _ = range time.Tick(5 * time.Second) {
-		kubevirtExist := checkForKubevirt(kubeClient)
-		log.V(1).Info("kubevirt exist in the cluster", "kubevirtExist", kubevirtExist)
-		if kubevirtExist {
-			log.Info("found kubevirt restarting the manager")
-			close(restartChannel)
-			break
-		}
-	}
-}
-
-// wait for the any interrupt to stop the manager and clean the webhook.
-func waitForSignal() {
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill)
-	<-c
-	close(restartChannel)
-}
-
 func main() {
+	var log logr.Logger
```

Comment:
I saw this on multiple PRs now and I need to understand the motivation. In some cases, we need to initialize `var` before defining it, but this is not the case. Do you initialize it here on purpose? Do you consider it better and why? For me, it is a waste of space and it has no added value, does it? I need to hear that there is a any motivation/intention for it, so I can move over.